javascript/src/main/components/Courses/CourseForm.js

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from "react";
+import { Form, Button, Row, Col, Container } from "react-bootstrap";
+
+const CourseForm = ({ createCourse, updateCourse, existingCourse }) => {
+    const emptyCourse = {
+        name: "",
+        quarter: "",
+        instructorFirstName: "",
+        instructorLastName: "",
+        instructorEmail: "",
+    }
+
+    const [course, setCourse] = useState(existingCourse || emptyCourse);
+
+    const handleOnSubmit = (e) => {
+        e.preventDefault();
+        if (createCourse)
+            createCourse(course.name, course.quarter, course.instructorFirstName, course.instructorLastName, course.instructorEmail);
```

Comment:
This should take a course object instead of spreading the parameters out.
```suggestion
            createCourse(course);
```